### ISSUE TITLE
 A dynamic way to update the style of TourProvider through hooks.

### ISSUE THREAD

**Issue by luo-justin** [elrumordelaluz/reactour#467]
I have enjoyed using the TourProvider feature with the useTour hook. However, I have encountered some limitations when applying styles dynamically when starting a new tour. For example, I have 2 different tours shown in my application, with 1 tour having certain dots being highlighted in the navigation bar whereas the second tour does not have any highlighted dots. Right now there is no way to update the dot CSS dynamically when using hooks. It would be nice to introduce a hook like setStyle to dynamically introduce a new style object to match a new step config.

**Describe the solution you'd like**
A setStyle hook to dynamically update the style object of the TourProvider.

**Describe alternatives you've considered**
Currently am only using css to adjust the styles. But not able to update the css to work with multiple tour configs since the css classes are named the same for the dot buttons each time.

**Additional context**
Basically, whenever we use `setSteps(newSteps) and setIsOpen(true)`, I want to be able to apply a new Style object like `setStyle(newStyle)` to match the new tour.

**Comment by elrumordelaluz** [comment]
Hi @luo-justin, thanks for open the _Issue_.

Mind sharing a minimal reproduction in a sandbox? Super curious about any possible solution.

**Comment by luo-justin** [comment]
I don't really have a solution to this problem and this is not any type of reproducible issue. I guess it's more of a feature request where we can dynamically change the style by introducing a `setStyle` hook that we use down in the component tree when using TourProvider. If you do know another way to dynamically change the style with hooks or with some other way, I am all ears. Thank you!

**Comment by elrumordelaluz** [comment]
I am asking a reproduction in order to understand where and how are you looking to update styles. 

Since each `step` [can update its own `style`](https://github.com/elrumordelaluz/reactour/tree/master/packages/tour#styles-stylesobj--popoverstylesobj--maskstylesobj), you could use [setSteps](https://github.com/elrumordelaluz/reactour/tree/master/packages/tour#setsteps-dispatchreactsetstateactionsteptype) to update styles, both for this specific step or for all/others.

[Here is a working example](https://codesandbox.io/s/tour-with-data-fetching-forked-ce1x2j?file=/src/index.js) which implements this:

```js
<button
  className="step2"
  onClick={() => {
    const color = randomColor();
    setSteps(
      steps.map((step, index) => {
        return index > 0
          ? {
              ...step,
              styles: {
                maskWrapper: (base) => ({ ...base, color }),
                badge: (base) => ({ ...base, backgroundColor: color })
              }
            }
          : step;
      })
    );
  }}
>
  Update styles
</button>
```

**Comment by s-pcode** [comment]
Hi @elrumordelaluz,

Can you please suggest how to change the popover background color on change of light mode to dark mode using custom theme. Updating all the steps for dark mode is not possible and it's a very long tour steps.

**Comment by elrumordelaluz** [comment]
@s-pcode did you try using the global [style](https://github.com/elrumordelaluz/reactour/tree/master/packages/tour#styles-stylesobj--popoverstylesobj--maskstylesobj-1) prop?

**Comment by s-pcode** [comment]
Yes I tired that but the color does not change when I toggle the dark mode. Below is the demo code ⬇️ 

```
popover: (base) => ({
 ...base,
borderRadius: 20,
background:
theme.palette.mode === "dark"? theme.palette.primary.main: theme.palette.error.main,
color: theme.palette.mode === "dark" ? "#fafafa" : "#212B36",
fontSize: "1rem",
})
```
It only applies the color initially but does not change the colors when dark mode is toggled.

**Comment by elrumordelaluz** [comment]
I think could be related on the context of `theme` and the position of _TourProvider_ in the tree. [Here](https://codesandbox.io/s/reactour-tour-demo-using-react-bootstrap-modal-forked-b6332k?file=/index.js) is simplified example that works as expected.

**Comment by s-pcode** [comment]
Thanks. I had to make a custom provider and it worked then. 👍

**Comment by elrumordelaluz** [comment]
I am closing this _issue_, but feel free to re-open or create a new one in any case.
